elements/classpower.lua

```diff
@@ -56,6 +62,32 @@ local function UpdateTexture(element)
 	end
 end
 
+local function UpdateStatusBar(element, full, partial, max)
+	for i = 1, max do
+		element[i]:Show()
+
+		if(i <= math.ceil(full + partial)) then
+			if(i <= full) then
+				element[i]:SetValue(1)
+			else
+				element[i]:SetValue(partial)
+			end
+		else
+			element[i]:SetValue(0)
+		end
+	end
+end
```

Comment:
What if the number of destro soul shards change through a talent or something?
If I don't want to show empty textures, i don't have to do anything layout wise. If I don't want to show empty bars, I'll have to hide them in PostUpdate. If I want to show empty class resources I can use the same background for both bars and textures. So why should one element behave like if it was two elements?